src/coreclr/jit/emitarm64.cpp

```diff
@@ -8629,6 +8675,25 @@ void emitter::emitIns_R_R(instruction     ins,
             }
             break;
 
+        case INS_sve_pmov:
+            if (sopt == INS_SCALABLE_OPTS_TO_PREDICATE)
```

Comment:
Could you add a check so that `INS_OPTS_SCALABLE_B` is required for `opt` (and all the other places `IF_SVE_CE_2A` and `IF_SVE_CF_2A` are checked)

This ensures the code calling it doesn't call with a different sized scalable. Eg: We want `theEmitter->emitIns_R_R(INS_sve_pmov, EA_SCALABLE, REG_P2, REG_V12, INS_OPTS_SCALABLE_H, INS_SCALABLE_OPTS_TO_PREDICATE);` to fail

Reply:
Actually, what would be even more useful is if `opt` is `INS_OPTS_SCALABLE_H/S/D` then it should just do:
`return emitIns_R_R_I(INS_sve_pmov, attr, reg1, reg2, 0, opt, sopt, INS_SCALABLE_OPTS_TO_PREDICATE);`

Why? Because it's a useful shortcut. Most of the time the code calling all of this will be wanting to do a pmov with immediate 0, and it's a pain to have to write:
`if (size == B) then emitIns_R_R(pmov,....) else emitIns_R_R_I(pmov,....,0);`